### ISSUE TITLE
Don't require an Xcode Runner target

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#56489]
Splitting out from https://github.com/flutter/flutter/issues/14648.

If you rename the Runner target, you get this error:
```
$ flutter build ios
Building com.example.testCreate for device (ios-release)...
The Xcode project does not define target "Runner" which is needed by Flutter tooling.
Open Xcode to fix the problem:
  open ios/Runner.xcworkspace
Encountered error while building for device.
```

In Xcode the scheme sets up which targets to build:
![Screen Shot 2020-05-06 at 3 17 19 PM](https://user-images.githubusercontent.com/682784/81233972-b3170a80-8fac-11ea-8c94-1a6170ae1ea6.png)

Flutter tooling uses `-scheme` for compiling the app, but `-target` to grab build settings.  Swap the build setting usage to scheme and remove the Runner target existence check.

tl;dr Flutter tooling requires a Runner target, which causes [pain for iOS developers](https://github.com/flutter/flutter/issues/14648), but that target isn't really necessary.
